### PR TITLE
Improve template object check

### PR DESCRIPTION
**Pull request by shvaikalesh**

### CODE REVIEW

**Comment by blia** [comment]
Thanks!
I know that my checks are very weak. :) It's just for PoC.
And I would not be able to make such a cool checking like you.
Could you apply the patch to my latest version, with keys support?

File: src/index.js [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-      dict.set(subj, description);
+      const desc = String.raw(first, ...keys.map(key => subj[key]));
+      dict.set(subj, desc);
       return subj;
     };
   }
-  const [subj] = args;
-  return dict.has(subj) ? dict.get(subj) : 'Undocumented subject';
+
+  return dict.get(first) || 'Undocumented subject';
```
Comment (shvaikalesh) [inline src/index.js:14]: I think it is reasonable to return `'Undocumented subject'` for falsy description.

File: src/index.js [hunk 2]
```diff
@@ -1,17 +1,17 @@
 const dict = new WeakMap();
+const isFrozenArray = val => Array.isArray(val) && Object.isFrozen(val);
+const isTemplateObject = val => isFrozenArray(val) && isFrozenArray(val.raw);
 
-export const doc = (...args) => {
-  const [strings, ...keys] = args;
-  if (Array.isArray(strings) && typeof strings.raw !== 'undefined') {
-
+export const doc = (first, ...keys) => {
```
Comment (shvaikalesh) [inline src/index.js:5]: Just wondering: is there a reason for having both named and default export?

**Comment by shvaikalesh** [comment]
BTW, have you considered pythonic comments?

```js
const add = (a, b) => {
  ``
    adds two numbers or concatenates two strings or whatever
  ``

  return a + b
}
```

`doc(add) // => "adds two numbers or concatenates two strings or whatever"`

File: src/index.js (hunk 1 above)
Comment (blia) [reply src/index.js:14]: it makes sense

File: src/index.js (hunk 2 above)
Comment (blia) [reply src/index.js:5]: I think yes. You able to have both names for one func in one module
```js
import describe, {doc} from '../src';

const user = describe
  `This is User object with id ${'id'}`
  (api.get('user', 1));

console.log(doc(user));
```

**Pull request merged**